jetty-websocket/javax-websocket-tests/src/main/java/org/eclipse/jetty/websocket/javax/tests/NetworkFuzzer.java

```diff
@@ -220,15 +222,34 @@ protected void handleException(Throwable failure)
     public static class FrameCapture implements FrameHandler
     {
         private final BlockingQueue<Frame> receivedFrames = new LinkedBlockingQueue<>();
-        private final EndPoint endPoint;
+        private EndPoint endPoint;
+        private final CompletableFuture<EndPoint> endPointFuture = new CompletableFuture<>();
         private final SharedBlockingCallback blockingCallback = new SharedBlockingCallback();
         private CoreSession coreSession;
 
-        public FrameCapture(EndPoint endPoint)
+        public void setEndpoint(EndPoint endpoint)
         {
-            this.endPoint = endPoint;
+            endPointFuture.complete(endpoint);
         }
 
+        private EndPoint getEndpoint() throws IOException
+        {
+            if (endPoint == null)
+            {
+                try
+                {
+                    endPoint = endPointFuture.get(5, TimeUnit.SECONDS);
+                    if (endPoint == null)
+                        throw new IllegalStateException();
+                }
+                catch (Exception e)
+                {
+                    throw new IOException(e);
+                }
+            }
+
+            return endPoint;
+        }
```

Comment:
Why this blocking code here?
I think you can work on the assumption that `setEndPoint()` (capital `P` please) is always called before any write, so just use an `EndPoint` field rather than a `CompletableFuture<EndPoint>`.

Reply:
This is just a test class, it is using the `customize(EndPoint)` to set the endpoint and then writes directly to the endpoint instead of going through the coreSession. So setEndPoint() is not always called before the write.

Reply:
I could wait for a for `FrameCapture` to open first which might be more simple